### PR TITLE
WIP refactor: code inspection - fix redundant local variable

### PR DESCRIPTION
**Pull request by zielint0**
WIP
DO NOT MERGE

### CODE REVIEW

File: src/main/java/spoon/refactoring/CtRenameLocalVariableRefactoring.java [hunk 1]
```diff
@@ -85,8 +85,7 @@ public void accept(CtReference t) {
 		@Override
 		public ScanningMode enter(CtElement element) {
 			if (ignoredParent != null && element instanceof CtElement) {
-				CtElement ele = (CtElement) element;
-				if (ele.hasParent(ignoredParent)) {
+				if (((CtElement) element).hasParent(ignoredParent)) {
```
Comment (surli) [inline src/main/java/spoon/refactoring/CtRenameLocalVariableRefactoring.java]: I really don't agree with this change: generally I'm the one extracting those casts in a local variable. I find it really more readable, and it's really useful when debugging. 

Now here I even don't understand why a cast is needed: element is already a CtElement, so we just need to remove the cast, don't we?

File: src/main/java/spoon/support/compiler/SnippetCompilationHelper.java [hunk 2]
```diff
@@ -50,13 +50,12 @@ private SnippetCompilationHelper() { }
 
 	public static void compileAndReplaceSnippetsIn(CtType<?> c) {
 		Factory f = c.getFactory();
-		CtType<?> workCopy = c;
 		Set<ModifierKind> backup = EnumSet.noneOf(ModifierKind.class);
-		backup.addAll(workCopy.getModifiers());
-		workCopy.removeModifier(ModifierKind.PUBLIC);
+		backup.addAll(c.getModifiers());
```
Comment (surli) [inline src/main/java/spoon/support/compiler/SnippetCompilationHelper.java]: Ok with the remove of workCopy but then could you rename the argument of the method: `c` is really not something explicit... `workType` maybe or whatever.

File: src/main/java/spoon/refactoring/CtRenameLocalVariableRefactoring.java (hunk 1 above)
Comment (zielint0) [reply src/main/java/spoon/refactoring/CtRenameLocalVariableRefactoring.java]: @surli 
> I really don't agree with this change: generally I'm the one extracting those casts in a local variable. I find it really more readable, and it's really useful when debugging.

Restored.

> Now here I even don't understand why a cast is needed: element is already a CtElement, so we just need to remove the cast, don't we?

We do :-)

File: src/main/java/spoon/refactoring/CtRenameLocalVariableRefactoring.java [hunk 3]
```diff
@@ -84,9 +84,9 @@ public void accept(CtReference t) {
 
 		@Override
 		public ScanningMode enter(CtElement element) {
-			if (ignoredParent != null && element instanceof CtElement) {
-				CtElement ele = (CtElement) element;
-				if (ele.hasParent(ignoredParent)) {
+			CtElement ele = element;
```
Comment (surli) [inline src/main/java/spoon/refactoring/CtRenameLocalVariableRefactoring.java]: I don't understand the purpose of local variable ele here: IMHO you can just remove this variable and the first if.

Comment (zielint0) [reply src/main/java/spoon/refactoring/CtRenameLocalVariableRefactoring.java]: @surli 
Done.

**Comment by zielint0** [comment]
WIP
DO NOT MERGE

**Pull request closed without merging**